### ISSUE TITLE
Crash when plotting ongoing measurement, related to pandas interpolation

### ISSUE THREAD

**Issue by abargerbos** [toolsforexperiments/plottr#23]
I'm running into a problem when I try to plot a specific type of VNA measurement; plottr will plot the first trace just fine, but once the second one begins it crashes. Opening it again still leads to a crash. Once the measurement has finished I can plot it just fine.

The error message I receive is
```
Traceback (most recent call last):
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 238, in setData
    self._plot2d(data, ax, axesNames[0], axesNames[1], dn)
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 197, in _plot2d
    im = ppcolormesh_from_meshgrid(ax, x, y, z)
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\plot\mpl.py", line 74, in ppcolormesh_from_meshgrid
    x, y = interp_meshgrid_2d(x, y)
  File "d:\onedrive\setups\lk1\lk1_code\code\plottr\plottr\utils\data.py", line 16, in interp_meshgrid_2d
    yy2 = pd.DataFrame(yy).interpolate(axis=0).values
  File "C:\ProgramData\Anaconda3\envs\qcodes\lib\site-packages\pandas\core\generic.py", line 6015, in interpolate
    raise TypeError("Cannot interpolate with all NaNs.")
TypeError: Cannot interpolate with all NaNs.
```

The measurement code used to generate the measurement is
```
setup_zerospan_trace(5.345737e9, 51)
      
sweep_obj = sweep(twpa_pump.power, np.arange(8.0, 12.0, .2))(
     sweep(twpa_pump.frequency, np.arange(7.8e9, 8.1e9, 1.0e6)))(
        measure(get_SNR)
)


result = do_experiment(f"TWPA_SNR/{SAMPLE}", sweep_obj, live_plot=False)
```

Here `get_SNR` is defined as 

```
@getter(('signal', 'dB'), ('noise', 'dB'), ('SNR', 'dB'))
def get_SNR():
    time.sleep(0.01)
    mag, phase = take_trace(plot=False)
    sig = np.abs(mag) * np.exp(-1j*phase)
    lin_mean = np.abs(sig.mean())
    lin_std = np.abs(sig.std())
    return 20*np.log10(lin_mean), 20*np.log10(lin_std), 20*np.log10(lin_mean/lin_std)
```

where `take_trace` is
```
def take_trace(chan='S21', plot=False):
    """
    Get the data of a currently measured trace.
    The trace has to be setup already.
    
    Returns magnitude (in dB) and phase (in rad).
    
    If plot is true, make a simple plot of the magnitude vs frequency.
    """
    vna = qc.Station.default.vna
    trace = getattr(vna.channels, chan)
    
    fvals = np.linspace(trace.start(), trace.stop(), trace.npts())
    mag, phase = trace.trace_mag_phase()
    
    if plot:
        fig, ax = plt.subplots(1, 1)
        ax.plot(fvals*1e-9, 20*np.log10(mag))
        ax.grid(dashes=[1,1])
        ax.set_xlabel('Frequency (GHz)')
        ax.set_ylabel('Magnitude (dBm)')
        
    return mag, phase
```

and `setup_zerospan_trace` is

```
def setup_zerospan_trace(frq, npts, chan='S21', span=1.):
    """
    Setup the VNA to measure a trace around <frq> with <npts> samples
    in a very narray frequency span.
    """
    vna = qc.Station.default.vna
    trace = getattr(vna.channels, chan)
    
    trace.center(frq)
    trace.span(span)
    trace.npts(npts)
    
    trace.autoscale()
    
    if not vna.rf_power():
        vna.rf_on()
```

**Comment by wpfff** [comment]
interesting one. i'll have a look into what might be happening, since it seems related to the sweep module.

**Comment by wpfff** [comment]
should hopefully be fixed with #39 -- waiting for confirmation.

**Comment by wpfff** [comment]
hey @abargerbos, is this fixed in your opinion? i'm closing it for now.
